packages/extension/src/components/molecules/NetworkIndicator/NetworkIndicator.tsx

```diff
@@ -92,18 +138,56 @@ export const NetworkIndicator: FC = () => {
   }, []);
 
   const handleClickOnNetwork = useCallback(
-    async (network: Network) => {
+    async (network: Network, customNetworkName?: string, customNetworkServer?: string) => {
       setIsLoading(true);
-      await switchNetwork(network);
+      await switchNetwork(network, customNetworkName, customNetworkServer);
+      setCurrentNetworkName(customNetworkName || network);
       handleClose();
     },
     [handleClose, switchNetwork]
   );
 
+  const handleAddNetworkClick = useCallback(() => {
+    setHandleAddNetwork(true);
+  }, []);
+
+  const handleAddNetworkClose = useCallback(() => {
+    setHandleAddNetwork(false);
+  }, []);
+
+  const removeNetwork = useCallback(
+    (networkName: string) => {
+      // If the network to delete is the active one, show the active network deletion warning dialog
+      if (networkName === currentNetworkName) {
+        setActiveNetworkDeleteDialogOpen(true);
+      } else {
+        setNetworkToDelete(networkName);
+        setConfirmDeleteOpen(true);
+      }
+    },
+    [currentNetworkName]
+  );
+
+  const handleConfirmDelete = useCallback(() => {
+    if (networkToDelete) {
+      try {
+        const updatedNetworks = { ...existingNetworks };
+        delete updatedNetworks[networkToDelete];
+        replaceCustomNetworks(updatedNetworks);
+        setExistingNetworks(updatedNetworks);
+      } catch (error) {
+        Sentry.captureException(error);
+      }
+    }
+
+    // close the confirmation dialog
+    setConfirmDeleteOpen(false);
+  }, [existingNetworks, networkToDelete]);
+
   return (
     <>
       <Chip
-        label={network || 'Switch network'}
+        label={currentNetworkName || network || 'Switch network'}
```

Comment:
```suggestion
        label={currentNetworkName || 'Switch network'}
```
I think this would be enough, because everytime you call setCurrentNetworkName you consider the `network` variable even at the beginning `currentNetworkName` is defaulted to `network`.

Reply:
Without it, it breaks the cypress tests